cli/loader_lib.c

```diff
@@ -163,6 +240,32 @@ __attribute__((constructor)) void jl_load_libjulia_internal(void) {
     // Introspect to find our own path
     const char * lib_dir = jl_get_libdir();
 
+#ifdef LIBSTDCXXPROBE
+    void* handle;
+    int do_probe = 1;
+    char* probevar = getenv("JLLIBSTDCXXPROBE");
+    if (probevar) {
+        if (strcmp(probevar, "1") == 0 || strcmp(probevar, "yes"))
+            do_probe = 1;
+        else if (strcmp(probevar, "0") == 0 || strcmp(probevar, "no"))
+            do_probe = 0;
+    }
+    if (do_probe) {
+        char cxxpath[JL_PATH_MAX];
+        size_t pathlen = 0;
+        if (!libstdcxxprobe(cxxpath, &pathlen)) {
+            jl_loader_print_stderr("ERROR: Syscalls failed while finding which libstdcxx path to load!\n");
+            exit(1);
+        }
+        handle = dlopen(cxxpath, RTLD_LAZY);
+        if (!handle) perror("dlopen"), exit(1);
+    }
+    else {
+        handle = dlopen("libstdc++.so.6", RTLD_LAZY);
```

Comment:
@staticfloat is this the correct file that we want to load unconditionally in this case?

DRY nit:
```suggestion
        handle = dlopen(JL_DEFAULT_STDCXXLIB, RTLD_LAZY);
```